core/domain/user_services.py

```diff
@@ -850,19 +922,70 @@ def create_new_user(gae_id, email):
     Raises:
         Exception. If a user with the given gae_id already exists.
     """
-    user_settings = get_user_settings(gae_id, strict=False)
+    user_settings = get_user_settings_by_gae_id(gae_id, strict=False)
```

Comment:
Great catch -- thanks @sajalasati! 

/cc @vojtechjelinek -- FYI this seems to have been missed during the migration. Can you confirm that this was an error?

Reply:
Yes, this is an error. Looking at the code this shouldn't have caused any problems since in base.py we use `get_user_settings_by_gae_id` for checking if the user exists and only then call  `create_new_user`, and here since `gae_id` and `user_id` have different lengths it will always return `None`. Also, this and lines 926 to 929 can probably be replaced by `get_user_settings_by_gae_id(gae_id, strict=True)`.

Reply:
@vojtechjelinek why `strict=True`? --this flag is meant for raising an exception when we don't find UserSettings. Here we want to raise an exception inside `create_new_user` when a UserSettings does exist. I don't think we can change how the strict flag is used either because we don't know inside `get_user_settings_by_gae_id` when to raise an error when UserSettings is found. I would prefer the current implementation only.

Reply:
Oh yeah, sorry my mistake, I just misread the code. The implementation is fine then.